lighthouse-core/computed/page-dependency-graph.js

```diff
@@ -388,23 +394,20 @@ class PageDependencyGraph {
     const networkNodeOutput = PageDependencyGraph.getNetworkNodeOutput(networkRecords);
     const cpuNodes = PageDependencyGraph.getCPUNodes(processedTrace);
 
-    // The root request is the earliest network request, using position in networkRecords array to break ties.
-    const rootRequest = networkRecords.reduce((min, r) => (r.startTime < min.startTime ? r : min));
-    const rootNode = networkNodeOutput.idToNodeMap.get(rootRequest.requestId);
     // The main document request is the earliest network request *of type document*.
     // This will be different from the root request when there are server redirects.
     const mainDocumentRequest = NetworkAnalyzer.findMainDocument(networkRecords);
     const mainDocumentNode = networkNodeOutput.idToNodeMap.get(mainDocumentRequest.requestId);
-
-    if (!rootNode || !mainDocumentNode) {
+    if (!mainDocumentNode) {
       // Should always be found.
-      throw new Error(`${rootNode ? 'mainDocument' : 'root'}Node not found.`);
+      throw new Error(`mainDocumentNode not found.`);
     }
 
-    if (mainDocumentNode !== rootNode &&
-        (!mainDocumentNode.record.redirects ||
-        !mainDocumentNode.record.redirects.includes(rootNode.record))) {
-      throw new Error('Root node was not in redirect chain of mainDocument');
+    // The root request is the earliest request in the main document redirect chain.
+    const rootRequest = mainDocumentNode.record.redirects && mainDocumentNode.record.redirects[0];
+    let rootNode = rootRequest && networkNodeOutput.idToNodeMap.get(rootRequest.requestId);
+    if (!rootNode) {
+      rootNode = mainDocumentNode;
```

Comment:
before when this was missing an error was thrown. should we do the same still?

Reply:
We will enter this condition in one of two cases:
- There were no redirects
- There was a redirect, but we didn't find a corresponding network node for it.

The first is handled correctly here. The second should probably throw. I'll make the changes.